SimTKcommon/src/Pathname.cpp

```diff
@@ -238,6 +261,43 @@ void Pathname::deconstructPathname( const string&   name,
     }
 }
 
+// We assume a path name structure like this:
+//   (1) Everything up to and including the final directory separator
+//       character is the directory; the rest is the file name. On return
+//       we fix the slashes in the directory name to suit the current
+//       system ('\' for Windows, '/' otherwise).
+//   (2) If the file name contains a ".", characters after the last
+//       "." are the extension and the last "." is removed.
+//   (5) What's left is the fileName.
+// We accept both "/" and "\" as separator characters. We leave the
+// case as it was supplied.
+// Leading and trailing white space is removed; embedded white space
+// remains.
+// Leading "X:" for some drive letter X is recognized on Windows as
+// a drive specification, otherwise the drive is the current drive.
+// That is then removed for further processing.
+// Absolute paths are designated like this:
+//      Leading "/" means root relative (on the drive).
+//      Leading "./" means current working directory relative (on the drive).
+//      Leading "../" is interpreted as "./..".
+//      Leading "@/" means relative to executable location.
+// Above leading characters are removed and replaced with the
+// full path name they represent, then the entire path is divided
+// into components. If a component is "." or "" (empty) it is
+// removed. If a component is ".." it and the previous component
+// if any are removed. (".." as a first component will report as
+// ill formed.)
+// If there is something ill-formed about the file name we'll return
+// false.
```

Comment:
Should this internal comment be moved out to the header so it shows in doxygen? It seems like it has some useful info that isn't given elsewhere.